### PR TITLE
Make it configurable

### PR DESCRIPTION
**Pull request by shikoku1**
Add toml configuration file, for example:

```
/etc/edgehog.conf
```
```toml
credentials_secret = "xxxxx"
device_id = "xxxxx"
pairing_url = "https://api.xxxxx/pairing"  
realm = "xxxxx"
interfaces_directory = "/var/lib/xxxx"
state_file = "/xxxx/xx"
download_directory = "/tmp"
```

### CODE REVIEW

File: src/config_file.rs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use edgehog_device_runtime::{error::DeviceManagerError, DeviceManagerOptions};
+use log::info;
+
+pub fn config_file_to_options(
+    custom_config_file_path: Option<String>,
+) -> Result<DeviceManagerOptions, DeviceManagerError> {
+    let paths = ["device-manager.conf", "/etc/device-manager.conf"]
```
Comment (bettio) [inline src/config_file.rs:27]: I would call this `/etc/edgehog.conf` instead.

File: src/config_file.rs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
```
Comment (bettio) [inline src/config_file.rs:20]: Maybe I would call this module just `config`.

File: src/config_file.rs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+use edgehog_device_runtime::{error::DeviceManagerError, DeviceManagerOptions};
+use log::info;
+
+pub fn config_file_to_options(
```
Comment (bettio) [inline src/config_file.rs:24]: what about `read_options()` ?

File: src/config_file.rs (hunk 1 above)
Comment (bettio) [reply src/config_file.rs:27]: what `custom_config_file_path` is meant to be?

**Review by bettio — CHANGES_REQUESTED** [review]

File: src/config_file.rs (hunk 1 above)
Comment (shikoku1) [reply src/config_file.rs]: > what `custom_config_file_path` is meant to be?

pass a custom path from the command line

Comment (shikoku1) [reply src/config_file.rs]: most programs have different paths for the config file, and a manual override
the first has priority

Comment (harlem88) [reply src/config_file.rs]: I think we should use two paths, a default and a custom path. If set the custom wins on the default. What do you think @bettio ?

File: README.md [hunk 4]
```diff
@@ -35,6 +35,20 @@ Edgehog Device Runtime is a reference implementation of
...
 
+## Configuration file
+Create a TOML configuration file either in /etc/edgehog.conf or ./edgehog.conf
+
+Example configuration:
+```toml
+credentials_secret = "xxxxx"
+device_id = "xxxxx"
+pairing_url = "https://api.xxxxx/pairing"
+realm = "xxxxx"
```
Comment (bettio) [inline README.md:46]: `examplerealm`

File: README.md [hunk 5]
```diff
@@ -35,6 +35,20 @@ Edgehog Device Runtime is a reference implementation of
...
+interfaces_directory = "/var/lib/xxxx"
```
Comment (bettio) [inline README.md:47]: `/usr/share/edgehog/astarte-interfaces/`

File: README.md [hunk 6]
```diff
@@ -35,6 +35,20 @@ Edgehog Device Runtime is a reference implementation of
...
+state_file = "/xxxx/xx"
```
Comment (bettio) [inline README.md:48]: `/var/lib/edgehog/state.json`

File: README.md [hunk 7]
```diff
@@ -35,6 +35,20 @@ Edgehog Device Runtime is a reference implementation of
...
+download_directory = "/tmp"
```
Comment (bettio) [inline README.md:49]: `/var/tmp/edgehog-updates`

File: README.md [hunk 8]
```diff
@@ -35,6 +35,20 @@ Edgehog Device Runtime is a reference implementation of
...
 features are implemented.
 
+## Configuration file
+Create a TOML configuration file either in /etc/edgehog.conf or ./edgehog.conf
+
+Example configuration:
+```toml
+credentials_secret = "xxxxx"
+device_id = "xxxxx"
+pairing_url = "https://api.xxxxx/pairing"
```
Comment (bettio) [inline README.md:45]: `https://api.astarte.example.com/pairing`

File: README.md [hunk 9]
```diff
@@ -35,6 +35,20 @@ Edgehog Device Runtime is a reference implementation of
 Astarte interfaces describe how data are exchanged with the remote instance, and what kind of
 features are implemented.
 
+## Configuration file
+Create a TOML configuration file either in /etc/edgehog.conf or ./edgehog.conf
+
+Example configuration:
+```toml
+credentials_secret = "xxxxx"
+device_id = "xxxxx"
```
Comment (bettio) [inline README.md:44]: device_id = "YOUR_UNIQUE_DEVIDE_ID"

File: README.md [hunk 10]
```diff
@@ -35,6 +35,20 @@ Edgehog Device Runtime is a reference implementation of
 Astarte interfaces describe how data are exchanged with the remote instance, and what kind of
 features are implemented.
 
+## Configuration file
+Create a TOML configuration file either in /etc/edgehog.conf or ./edgehog.conf
+
+Example configuration:
+```toml
+credentials_secret = "xxxxx"
```
Comment (bettio) [inline README.md:43]: YOUR_CREDENTIAL_SECRET

File: README.md [hunk 11]
```diff
@@ -35,6 +35,20 @@ Edgehog Device Runtime is a reference implementation of
 Astarte interfaces describe how data are exchanged with the remote instance, and what kind of
 features are implemented.
 
+## Configuration file
+Create a TOML configuration file either in /etc/edgehog.conf or ./edgehog.conf
```
Comment (bettio) [inline README.md:39]: `Edgehog Device Runtime can be configured using a [TOML](https://en.wikipedia.org/wiki/TOML) located either in $PWD/edgehog.conf or  /etc/edgehog.conf.`

**Review by bettio — CHANGES_REQUESTED** [review]

File: README.md (hunk 11 above)
Comment (harlem88) [inline README.md:39]: describe also the `command line` option.

**Review by harlem88 — CHANGES_REQUESTED** [review]

**Review by harlem88 — APPROVED** [review]

File: README.md [hunk 12]
```diff
@@ -35,6 +35,21 @@ Edgehog Device Runtime is a reference implementation of
 Astarte interfaces describe how data are exchanged with the remote instance, and what kind of
 features are implemented.
 
+## Configuration file
```
Comment (bettio) [inline README.md:38]: Let's just write "## Configuration"

File: README.md [hunk 13]
```diff
@@ -35,6 +35,21 @@ Edgehog Device Runtime is a reference implementation of
...
+
+Edgehog Device Runtime can be configured using a [TOML](https://en.wikipedia.org/wiki/TOML) located either in $PWD/config.toml or /etc/edgehog/config.toml, or in a custom path, run `cargo run -- --help` for more informations.
```
Comment (bettio) [inline README.md:40]: `using a [TOML](https://en.wikipedia.org/wiki/TOML) file located`

File: README.md [hunk 14]
```diff
@@ -35,6 +35,21 @@ Edgehog Device Runtime is a reference implementation of
...
+
+Example configuration:
+```toml
+credentials_secret = "YOUR_CREDENTIAL_SECRET"
+device_id = "YOUR_UNIQUE_DEVIDE_ID"
+pairing_url = "https://api.astarte.example.com/pairing"
```
Comment (bettio) [inline README.md:46]: example.com -> EXAMPLE.COM

File: README.md [hunk 15]
```diff
@@ -35,6 +35,21 @@ Edgehog Device Runtime is a reference implementation of
...
+realm = "examplerealm"
+interfaces_directory = "/usr/share/edgehog/astarte-interfaces/"
+state_file = "/var/lib/edgehog/state.json"
+download_directory = "/var/tmp/edgehog-updates"
```
Comment (bettio) [inline README.md:50]: let's keep ourself consistent, do we want to write path with our without trailing `/`?

**Review by bettio — CHANGES_REQUESTED** [review]

**Pull request merged**